ambry-router/src/main/java/com.github.ambry.router/GetOperation.java

```diff
@@ -41,10 +41,10 @@
   protected final NonBlockingRouterMetrics routerMetrics;
   protected final ClusterMap clusterMap;
   protected final ResponseHandler responseHandler;
-  protected final FutureResult<GetBlobResult> operationFuture;
-  protected final Callback<GetBlobResult> operationCallback;
+  protected final FutureResult operationFuture;
```

Comment:
I have created a `GetBlobResultInternal` class for all types of results, so this brings back the typed future and callbacks. The instance itself will have the information about what kind of result it is.